plugins/outputs/application_insights/transmitter.go

```diff
@@ -1,13 +1,22 @@
 package application_insights
 
-import "github.com/Microsoft/ApplicationInsights-Go/appinsights"
+import (
+	"github.com/Microsoft/ApplicationInsights-Go/appinsights"
+)
 
 type Transmitter struct {
 	client appinsights.TelemetryClient
 }
 
-func NewTransmitter(ikey string) *Transmitter {
-	return &Transmitter{client: appinsights.NewTelemetryClient(ikey)}
+func NewTransmitter(ikey string, EndpointUrl string) *Transmitter {
+
+	if len(EndpointUrl) == 0 {
```

Comment:
I’d merge this with the code below and have the conditional only around the endpoint url setter

Reply:
Please check it. Did i understand you correctly